MetaMask/metamask-extension
# fix: storybook theme package not working

## **Description**

This PR removes the `storybook-dark-mode` package and utilizes the native Storybook Theme API for managing the light and dark modes across both the documentation and component preview pages.

The motivation behind this change is to fix the bug introduced after upgrading Storybook to ^7.6.20 and `storybook-dark-mode` to ^4.0.2, which resulted in the following error:

```
Error: Storybook preview hooks can only be called inside decorators and story functions.
```

Additionally, we have updated the `DocsContainer` to ensure it handles the theme correctly without throwing errors. The bug prevented Storybook Docs pages from rendering properly.

Key changes:
1. Removed the `storybook-dark-mode` package.
2. Implemented the native Storybook Theme API to manage light/dark modes.
3. Added a button to toggle between light, dark and side by side modes in the component preview area.
4. Ensured the theme context is properly checked and applied, particularly in the Docs rendering.

Additional Context:
- We will be implementing the shared UI Component library soon and we can revisit whether having the additional add-on (`storybook-dark-mode`) is valuable enough to add an additional dependency. We can add it to our roadmap for discussion when we are working on the storybook portion of our roadmap.

## Possible Asked Questions
- Did you consider that the app theme is not changing? 
Yes. It is set to the users system, so if the user uses dark mode then the application will run in dark. Made attempts for a fix 
- Did you consider a fix with the existing package? 
Yes. The best solution was to revert to the old package version, but that stops us from progressing forward. It's nice to remove dependencies and use built-in options.

 
## **Related issues**

Fixes: 

- #26703 (Introduced the error after upgrading to Storybook 7.6.20)

## **Manual testing steps**

1. Run `yarn storybook`.
2. Visit the [component docs page running locally](http://localhost:6006/?path=/docs/components-componentlibrary-avataraccount--docs).
3. Verify that no errors related to theme switching appear in the console.
4. Ensure the docs page renders properly in both light and dark themes.
5. Test theme switching using the button added in the component preview.

## **Screenshots/Recordings**

### **Before**

![Error Screenshot](https://github.com/user-attachments/assets/fcd39dbb-6bbd-49c5-8974-e446c491ee26)


https://github.com/user-attachments/assets/2699deff-1f8e-4900-a982-07ff5918121c


### **After**

This now allows us to have more than one theme for our components and we can also view both modes at the same time! 



https://github.com/user-attachments/assets/f2588aae-5c7c-49de-91f6-41e2fa86be42



Here is where I know some people might be upset, but currently the entire application theme is set to users system and no longer tied to the theme switch

https://github.com/user-attachments/assets/080c3cf4-669e-4781-8eb6-af2688fd7589



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pulled and built the branch, run the app, tested code being changed).
